k_retrieval function to find the k obj of a list with more similarity…

mrfarge opened the pull request:
… with the query

mrfarge committed:
k_retrieval function to find the k obj of a list with more similarity with the query

hermesvf commented on src/k_retrieval.py:
```diff
@@ -0,0 +1,31 @@
+def k_retrieval(distances_list, similarity_methods, k):
```
It is not clear to me what do each argument represent (other than `k`).

hermesvf commented on src/k_retrieval.py:
```diff
@@ -0,0 +1,31 @@
+    
+    """
+    
+    method=distances_list[0].method
```
why is it a list and not a single distance?

mrfarge replied:
I've assumed a list of distances as input, so I think it's easier to rank the distance. Do you think would be better do it different?

hermesvf commented on src/k_retrieval.py:
```diff
@@ -0,0 +1,31 @@
+    
+    method=distances_list[0].method
+
+    if method == "euclidean" or method == "L1_dist" or method == "x2_dist" or method == "hellinger":
```
The `if` and the `else` have almost the same code. I would change to something like 
```suggestion
    maximization_required = method not in [ "euclidean", "L1_dist", "x2_dist", "hellinger"]
    distances_list.sort(key=lambda x: x.distance, reverse=maximization_required)
    k_distances=distances_list[0 : k+1]
    

```

hermesvf replied:
Even better, I would store in the "distance" class if the metric needs maximization or minimization.

mrfarge replied:
Yep, you're right, I'm changing it.

mrfarge committed:
Update k_retrieval.py

k_retrieval optimized, similarity methods not necessaries as it's stored in the class distance
